python/cudf_polars/tests/test_config.py

```diff
@@ -0,0 +1,32 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+import os
+
+import polars as pl
+
+from cudf_polars.dsl.expr import Expr
+import cudf._lib.pylibcudf as plc
+
+
+class NotImplementedExpr(Expr):
+    def do_evaluate(self, df, *, context, mapping):
+        raise NotImplementedError("Example Unsupported cuDF operation")
+
+
+def invoke_unimplemented_op(df):
+    # TODO: does not work
+    # df.select(NotImplementedExpr(plc.DataType(plc.TypeId.INT64)))
```

Comment:
@wence- do you think this is on the right track for how to test this, and if so can you point me in the direction of how to invoke a custom expr against a dataframe like this?

Reply:
I was going to review this PR but unfortunately I don't know enough to answer this question so I'll have to defer to Lawrence.

Reply:
This was the first place my thoughts went as well, but eventually we will have scans, and then we'd need to update the test. Can't we come up with a generic testing expression we know will never be implemented, and use that?

Reply:
Good point, we can do this by monkeypatching `__post_init__` of the IR node:
```
def test_with_monkeypatch(monkeypatch):
    from cudf_polars.dsl.ir import IR
    def raises(self):
        raise NotImplementedError("Oh no!")
    monkeypatch.setattr(IR, "__post_init__", lambda self: raise )

    q = pl.LazyFrame({})
    with pl.Config(verbose=True), pytest.warns(...):
        assert_ir_translation_raises(q, NotImplementedError)
```

Reply:
If fixed my suggestion and pushed.

Reply:
Thanks Lawrence!